app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala

```diff
@@ -18,60 +18,58 @@ package uk.gov.hmrc.helptosavefrontend.controllers
 
 import javax.inject.Singleton
 
-import cats.data.{EitherT, OptionT, ValidatedNel}
+import cats.data.{EitherT, ValidatedNel}
 import cats.instances.future._
 import cats.syntax.either._
 import com.google.inject.Inject
-import play.api.Logger
 import play.api.i18n.{I18nSupport, MessagesApi}
-import play.api.mvc.Action
-import uk.gov.hmrc.helptosavefrontend.auth.HtsCompositePageVisibilityPredicate.twoFactorURI
+import play.api.mvc.{Action, AnyContent}
+import play.api.{Application, Logger}
 import uk.gov.hmrc.helptosavefrontend.connectors.NSIConnector.{SubmissionFailure, SubmissionResult, SubmissionSuccess}
 import uk.gov.hmrc.helptosavefrontend.connectors._
 import uk.gov.hmrc.helptosavefrontend.models.{HTSSession, NSIUserInfo, UserInfo}
-import uk.gov.hmrc.helptosavefrontend.models.UserInfo
-import uk.gov.hmrc.helptosavefrontend.services.userinfo.UserInfoService
+import uk.gov.hmrc.helptosavefrontend.services.HelpToSaveService
 import uk.gov.hmrc.helptosavefrontend.util.Result
 import uk.gov.hmrc.helptosavefrontend.views
-import uk.gov.hmrc.play.frontend.auth.AuthContext
-import uk.gov.hmrc.play.frontend.auth.connectors.domain.Accounts
 import uk.gov.hmrc.play.http.HeaderCarrier
 
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class RegisterController @Inject()(val sessionCacheConnector: SessionCacheConnector,
-                                   val messagesApi: MessagesApi,
-                                   eligibilityConnector: EligibilityConnector,
-                                   citizenDetailsConnector: CitizenDetailsConnector,
-                                   nSAndIConnector: NSIConnector) extends HelpToSaveController with I18nSupport {
-
-  val userInfoService = new UserInfoService(authConnector, citizenDetailsConnector)
-
-  def declaration =
-    AuthorisedHtsUserAction { implicit authContext ⇒
-      implicit request ⇒
-        validateUser(authContext).fold(
+class RegisterController @Inject()(override val messagesApi: MessagesApi,
+                                   htsService: HelpToSaveService,
+                                   nSAndIConnector: NSIConnector,
+                                   val sessionCacheConnector: SessionCacheConnector,
+                                   implicit val app: Application)
+  extends HelpToSaveAuth(app) with I18nSupport {
+
+  def declaration: Action[AnyContent] = authorisedForHtsWithEnrolments {
+    implicit request ⇒
+      implicit userUrlWithNino ⇒
+        val userInfo = for {
+          userUrlWithNino <- EitherT.fromOption[Future](userUrlWithNino, "could not retrieve either userDetailsUrl or NINO from auth")
+          result <- checkEligibility(userUrlWithNino.path, userUrlWithNino.nino)
```

Comment:
Also, I don't think `HTSSession` should have an `Option[UserInfo]` in it, it should have just a `UserInfo` in it - why would we want to write `None` to keystore? But this isn't part of this PR